### ISSUE TITLE
[$20] If the user select an end time first, the same time can be selected in the start time

### ISSUE THREAD

**Issue by sdgun** [topcoder-platform/taas-app#542]
**Steps**

- Select 'Schedule interview' action from CANDIDATES TO REVIEW page for a candidate
- Click Add availability
- Click on a date(ex-M)
- Select an end time and select the same time from start time

If the user select an end time first, the same time can be selected in the start time

![image](https://user-images.githubusercontent.com/4592712/139378879-046077fd-6dbc-492c-9c16-935ebbd619fb.png)

**Comment by maxceem** [comment]
> In addition to the above, the user is able to choose the same day and same time multiple times.

@sandhiyakavi I'm not sure what should be desired behaviour in this case? should we show validation error? I just think that showing validation error in such case might make the UI more cluttered, as there is already many elements in UI there. So maybe we just allow use to add rows with the same data? I think it should not bring any issue.
Or is there any good way to handle this case?

**Comment by sandhiyakavi** [comment]
@maxceem Yes, I expected some validation error but since it will complicate things and functionality wise there is no problem even if it is duplicated. We can allow adding rows with the same data

**Comment by maxceem** [comment]
Sum up:
- Only the part which is mentioned in the issue description is in scope https://github.com/topcoder-platform/taas-app/issues/542#issue-1039175590
- we **should not** allow selecting the same time in start time and end time, options has to be disable
- exception: we **should be** able to select 12:00 AM in start time and 12:00 AM in end time - it would mean any time

**Comment by maxceem** [comment]
@yoution feel free to pick up if you like.

**Comment by yoution** [comment]
@maxceem please assign to me

**Comment by sdgun** [comment]
Verified in Dev, this issue is fixed.

**Comment by maxceem** [comment]
Challenge https://www.topcoder.com/challenges/e5ecf392-81c9-4cfc-9d84-9bd168f96e4a has been created for this ticket.<br/><br/>```This is an automated message for maxceem via Topcoder X```

**Comment by maxceem** [comment]
Challenge https://www.topcoder.com/challenges/e5ecf392-81c9-4cfc-9d84-9bd168f96e4a has been assigned to yoution.<br/><br/>```This is an automated message for maxceem via Topcoder X```

**Comment by maxceem** [comment]
Payment task has been updated: https://www.topcoder.com/challenges/e5ecf392-81c9-4cfc-9d84-9bd168f96e4a
*Payments Complete*
Winner: yoution
Copilot: maxceem
Challenge `e5ecf392-81c9-4cfc-9d84-9bd168f96e4a` has been paid and closed.<br/><br/>```This is an automated message for maxceem via Topcoder X```
